pytests/test_transforms.py

```diff
@@ -0,0 +1,45 @@
+import numpy as np
+import pytest
+from numpy.testing import assert_array_almost_equal
+
+from pylops.transforms import (
+    DiscreetCosine
+)
+from pylops.utils import dottest
+
+par1 = {"ny": 11, "nx": 11, "imag": 0, "dtype": "float64"}
+par2 = {"ny": 11, "nx": 21, "imag": 0, "dtype": "float64"}
+par3 = {"ny": 21, "nx": 21, "imag": 0, "dtype": "float64"}
+
+@pytest.mark.parametrize("par", [(par1), (par3)])
+def test_DiscreetCosine1D(par):
+    """Dot test  for Discreet Cosine Transform Operator 1D"""
+
+    t = np.arange(par["ny"])
+
+    Dct = DiscreetCosine(dims=(par["ny"],), dtype=par["dtype"])
+
+    assert dottest(Dct, par["ny"], par["ny"], rtol=1e-6, complexflag=0, verb=True)
+    
+    y=Dct.H*(Dct*t)
+
+    assert_array_almost_equal(t, y, decimal=3)
+
+
+@pytest.mark.parametrize("par", [(par1), (par2), (par3)])
+def test_DiscreetCosine2D(par):
+    """Dot test  for Discreet Cosine Transform Operator 2D"""
+
+    t = np.outer(np.arange(par["ny"]), np.arange(par["nx"]))
+
+    Dct = DiscreetCosine(dims=t.shape, dtype=par["dtype"])
+
+    assert dottest(Dct, par["nx"]*par["ny"], par["nx"]*par["ny"], rtol=1e-6, complexflag=0, verb=True)
+
+    y = Dct.H*(Dct*t)
+
+    assert_array_almost_equal(t, y, decimal=3)
+
+
+    
```

Comment:
We should add a test where DCT is applied to a 2d signal over one axis, and to a nd signal (eg 3d) over 2 axes

Reply:
added more test cases with additional parameters and 3D test case as well in [1542bfc](https://github.com/PyLops/pylops/pull/480/commits/1542bfcb2814a2d7a90ab9eabf72ea99b6c8f597)